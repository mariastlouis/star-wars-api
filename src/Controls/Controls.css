@import url("https://fonts.googleapis.com/css?family=Pathway+Gothic+One");
@import url("https://fonts.googleapis.com/css?family=Cabin");
body {
  background: black;
  color: white;
  font-family: "Cabin", sans-serif; }

h1 {
  font-size: 20px; }

h2 {
  display: block;
  -webkit-margin-before: 0;
  -webkit-margin-after: 0;
  -webkit-margin-start: 0px;
  -webkit-margin-end: 0px; }

button.compare-button {
  font-size: 15px;
  margin: 10px;
  padding: 10px 15px; }

button.compare-button:hover {
  background: #feda4a; }

.controls {
  display: flex;
  justify-content: center; }

.active {
  background: #feda4a; }
