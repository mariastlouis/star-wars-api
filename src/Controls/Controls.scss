@import 'variable';

button.compare-button {
  font-size: 15px;
  margin: 10px;
  padding: 10px 15px;
}

button.compare-button:hover{
  background: $accent-color;
}

.controls {
  display: flex;
  justify-content: center;
}

.active {
  background: $accent-color;
}