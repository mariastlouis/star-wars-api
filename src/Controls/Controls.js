import React from 'react';
import PropTypes from 'prop-types';
import './Controls.css';


export default class Controls extends React.Component {

  constructor() {
    super();
    this.state = {active: 0};
    this.passClickInfo = this.passClickInfo.bind(this);
  }

  passClickInfo (category, position) {
    this.props.clickCategory(category);
    this.toggle(position);
  }

  toggle(position) {
    if (this.state.active === position) {
      this.setState({active : null});
    } else {
      this.setState({active: position});
    }
  }

  bgColor(position) {
    if (this.state.active === position) {
      return "#feda4a";
    } return "";
  }

  getActive(position) {
    if(this.state.active === position) {
      return 'active'
    } return ''
  }

 

  render(){
    const favoriteNumber = this.props.favorites.length;
    return (
      <div className = "controls">
        <button 
          // style = {{backgroundColor: this.bgColor(0)}}
          className={`compare-button character ${this.getActive(0)}`}
          onClick={ () => this.passClickInfo('character', 0)}>
            People
        </button>
        <button
          style = {{backgroundColor: this.bgColor(1)}} 
          className={`compare-button planet ${this.getActive(1)}`}
          onClick={ () => this.passClickInfo('planet', 1)}>
            Planets
        </button>
        <button
          style = {{backgroundColor: this.bgColor(2)}}
          // className="compare-button vehicle"
          className={`compare-button vehicle ${this.getActive(2)}`}
          onClick={ () => this.passClickInfo('vehicle', 2)}>
            Vehicles
        </button>
        <button
          style = {{backgroundColor: this.bgColor(3)}}
          className="compare-button favorite"
          onClick={ () => this.passClickInfo('favorite', 3)}>
            Favorites: {favoriteNumber}
        </button>
      </div>
    );
  }
}

Controls.propTypes = {
  clickCategory: PropTypes.func,
  favorites: PropTypes.array
};

