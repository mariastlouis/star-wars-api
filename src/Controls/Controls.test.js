import React from 'react';
import Controls from './Controls.js';
import {shallow} from 'enzyme';

const mockClickFunction = jest.fn();

const mockFavorites = [
    {
      name: 'Sand Crawler',
      favorite: false,
      info: {
        Class: 'Wheeled',
        Model: 'Digger Crawler',
        Passengers: '30'
      }
    },
    {
      name: "X-34 landspeeder",
      favorite: true,
      info: {
        Class: "repulsorcraft",
        Model: "X-34 landspeeder",
        Passengers: "1"
      }

    }
  ]

let wrapper;

describe('Controls', () =>{
  beforeEach(() =>{
    wrapper = shallow (
      <Controls
         clickCategory = {mockClickFunction}
         favorites = {mockFavorites}
      />);
  })

  it('should render correctly', () =>{
    expect(wrapper).toBeDefined();
  })

  it('should have four buttons', () =>{
    expect(wrapper.find('button').length).toEqual(4);
  });

  it('should match the snapshot', () =>{
    expect(wrapper).toMatchSnapshot();
  });

  it('should call the click function if a button is pressed', () =>{
    expect(mockClickFunction.mock.calls.length).toEqual(0);
    wrapper.find('.planet').simulate('click');
    expect(mockClickFunction.mock.calls.length).toEqual(1);
     wrapper.find('.character').simulate('click');
    expect(mockClickFunction.mock.calls.length).toEqual(2);
 });

  it('should have 2 favorites listed on the favorite button if two favorite objects are passed in', () =>{
    expect(wrapper.find('.favorite').text()).toEqual('Favorites: 2')
  })



});