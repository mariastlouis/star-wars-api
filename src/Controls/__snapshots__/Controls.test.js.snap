// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Controls
    clickCategory={[Function]}
    favorites={
        Array [
            Object {
              "favorite": false,
              "info": Object {
                "Class": "Wheeled",
                "Model": "Digger Crawler",
                "Passengers": "30",
              },
              "name": "Sand Crawler",
            },
            Object {
              "favorite": true,
              "info": Object {
                "Class": "repulsorcraft",
                "Model": "X-34 landspeeder",
                "Passengers": "1",
              },
              "name": "X-34 landspeeder",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="compare-button character"
          onClick={[Function]}
          style={
                    Object {
                              "backgroundColor": "#feda4a",
                            }
          }
>
          People
</button>,
        <button
          className="compare-button planet"
          onClick={[Function]}
          style={
                    Object {
                              "backgroundColor": "",
                            }
          }
>
          Planets
</button>,
        <button
          className="compare-button vehicle"
          onClick={[Function]}
          style={
                    Object {
                              "backgroundColor": "",
                            }
          }
>
          Vehicles
</button>,
        <button
          className="compare-button favorite"
          onClick={[Function]}
          style={
                    Object {
                              "backgroundColor": "",
                            }
          }
>
          Favorites: 
          2
</button>,
      ],
      "className": "controls",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "People",
          "className": "compare-button character",
          "onClick": [Function],
          "style": Object {
            "backgroundColor": "#feda4a",
          },
        },
        "ref": null,
        "rendered": "People",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Planets",
          "className": "compare-button planet",
          "onClick": [Function],
          "style": Object {
            "backgroundColor": "",
          },
        },
        "ref": null,
        "rendered": "Planets",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Vehicles",
          "className": "compare-button vehicle",
          "onClick": [Function],
          "style": Object {
            "backgroundColor": "",
          },
        },
        "ref": null,
        "rendered": "Vehicles",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Favorites: ",
            2,
          ],
          "className": "compare-button favorite",
          "onClick": [Function],
          "style": Object {
            "backgroundColor": "",
          },
        },
        "ref": null,
        "rendered": Array [
          "Favorites: ",
          2,
        ],
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="compare-button character"
            onClick={[Function]}
            style={
                        Object {
                                    "backgroundColor": "#feda4a",
                                  }
            }
>
            People
</button>,
          <button
            className="compare-button planet"
            onClick={[Function]}
            style={
                        Object {
                                    "backgroundColor": "",
                                  }
            }
>
            Planets
</button>,
          <button
            className="compare-button vehicle"
            onClick={[Function]}
            style={
                        Object {
                                    "backgroundColor": "",
                                  }
            }
>
            Vehicles
</button>,
          <button
            className="compare-button favorite"
            onClick={[Function]}
            style={
                        Object {
                                    "backgroundColor": "",
                                  }
            }
>
            Favorites: 
            2
</button>,
        ],
        "className": "controls",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "People",
            "className": "compare-button character",
            "onClick": [Function],
            "style": Object {
              "backgroundColor": "#feda4a",
            },
          },
          "ref": null,
          "rendered": "People",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Planets",
            "className": "compare-button planet",
            "onClick": [Function],
            "style": Object {
              "backgroundColor": "",
            },
          },
          "ref": null,
          "rendered": "Planets",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Vehicles",
            "className": "compare-button vehicle",
            "onClick": [Function],
            "style": Object {
              "backgroundColor": "",
            },
          },
          "ref": null,
          "rendered": "Vehicles",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Favorites: ",
              2,
            ],
            "className": "compare-button favorite",
            "onClick": [Function],
            "style": Object {
              "backgroundColor": "",
            },
          },
          "ref": null,
          "rendered": Array [
            "Favorites: ",
            2,
          ],
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
